### PR TITLE
[PW-3952] Validate cart value and currency before each /payments and /payments/details request

### PR DESCRIPTION
**Pull request by cyattilakiss**
## Summary
Add ADYEN_OS_PAYMENT_NEEDS_ATTENTION new order status
Alter adyen_payment_response to add request_amount and request_currency 
Create waiting for payment orders for RedirectShopper methods
Avoid creating order in case the cart amount or currency has changed during the payment process
In case the payment is verified and the cart has changed, flag the order as needs attention

**Fixed issue**:
#151 [PW-3952] Bank amount different of prestashop total paid amount

### CODE REVIEW

File: adyenofficial.php [hunk 1]
```diff
@@ -381,6 +392,43 @@ public function createWaitingForPaymentOrderStatus()
         return true;
     }
 
+    /**
+     * Create a new order status: "waiting for payment"
```
Comment (Morerice) [inline adyenofficial.php:398]: Please update the comment based on the function

Comment (cyattilakiss) [reply adyenofficial.php]: updated!

**Review by Morerice — DISMISSED** [review]

File: controllers/FrontController.php [hunk 2]
```diff
@@ -226,6 +252,16 @@ protected function handlePaymentsResponse($response, $cart, $customer, $isAjax)
                 // PaymentResponse can be deleted
                 $this->adyenPaymentResponseModel->deletePaymentResponseByCartId($cart->id);
 
+                if ($cart->OrderExists() !== false) {
+                    $order = $this->orderAdapter->getOrderByCartId($cart->id);
+                    if (\Validate::isLoadedObject($order)) {
+                        //$orderPayment = \OrderPaymentCore::getByOrderReference($order->reference);
```
Comment (cyattilakiss) [inline controllers/FrontController.php:258]: Not use can be removed
```suggestion
```

**Review by Morerice — APPROVED** [review]

File: controllers/FrontController.php [hunk 3]
```diff
@@ -254,14 +289,26 @@ protected function handlePaymentsResponse($response, $cart, $customer, $isAjax)
 
                 break;
             case 'RedirectShopper':
-                // When the resultCode is RedirectShopper the cart needs to be cleared
-                $this->context->cookie->__set("id_cart", "");
-                // Continue with the same logic as IdentifyShopper and ChallengeShopper
+                // Create an ordder for each redirectShopper payments with the state of ADYEN_OS_WAITING_FOR_PAYMENT
```
Comment (acampos1916) [inline controllers/FrontController.php:292]:
```suggestion
                // Create an order for each redirectShopper payments with the state of ADYEN_OS_WAITING_FOR_PAYMENT
```

**Review by acampos1916 — APPROVED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Review by acampos1916 — APPROVED** [review]

**Review by Morerice — APPROVED** [review]

**Pull request merged**